Sources/TuistAcceptanceTesting/TuistAcceptanceTestCase.swift

```diff
@@ -0,0 +1,141 @@
+// swiftlint:disable force_try
+import TSCBasic
+import TuistCore
+import TuistGraph
+import TuistSupportTesting
+import XCTest
+
+@testable import TuistKit
+@testable import TuistSupport
+
+public enum Destination {
+    case simulator, device
+}
+
+open class TuistAcceptanceTestCase: TuistTestCase {
+    public var xcodeprojPath: AbsolutePath!
+    public var workspacePath: AbsolutePath!
+    public var fixturePath: AbsolutePath!
+    public var derivedDataPath: AbsolutePath!
+    public var cacheDirectory: AbsolutePath!
+
+    private var sourceRootPath: AbsolutePath!
+
+    override open func setUp() {
+        super.setUp()
+
+        cacheDirectory = try! TemporaryDirectory(removeTreeOnDeinit: true).path
+        derivedDataPath = try! TemporaryDirectory(removeTreeOnDeinit: true).path
+        setenv(
```

Comment:
Isn't this `setenv` setting a global state that might lead to state leaking to other tests that are running in parallel, potentially causing them to be flaky?

Reply:
I'd not create a cache and derived directory per test. Since we create a temporary directory per project, we should rely on Xcode and `xcodebuild` doing the right isolations in derived data. And the same is true for the `cache` directory, which Tuist uses for warming. If our fingerprinting logic is right, parallelism shouldn't have any negative impact on the test execution.

Reply:
> I'd not create a cache and derived directory per test. Since we create a temporary directory per project, we should rely on Xcode and xcodebuild doing the right isolations in derived data.

It's useful to control both to do assertions on the contents of those directories. That's how we did it in cucumber.

Reply:
> Isn't this setenv setting a global state that might lead to state leaking to other tests that are running in parallel, potentially causing them to be flaky?

Parallel tests are run as their own process, so there's no shared state

Reply:
> Parallel tests are run as their own process, so there's no shared state

I added the following piece of code:

```swift
let valueBefore = getenv(Constants.EnvironmentVariables.forceConfigCacheDirectory)
if let valueBefore = valueBefore {
    print("VALUE_BEFORE: \(valueBefore)")
}
```

<img width="602" alt="image" src="https://github.com/tuist/tuist/assets/663605/1fde6c0f-f37b-4767-9ef9-023da3473ca2">

And hit it a few times while running all the tests in parallel, which makes me think that the state is shared, otherwise I shouldn't hit that piece of code at all. I think system-wide is the same process, and `XCTest` uses creates the concurrency within the same process.

I'm afraid this will require some refactoring to pass that information down from the command level. Either that or we merge this one with the parallelism disabled and do the refactor in a follow-up PR.